__tests__/posts.ts

```diff
@@ -1916,3 +1923,106 @@ describe('mutation submitExternalLink', () => {
     expect(sharedPost2?.visible).toEqual(false);
   });
 });
+
+describe('mutation checkLinkPreview', () => {
+  const MUTATION = `
+    mutation CheckLinkPreview($url: String!) {
+      checkLinkPreview(url: $url) {
+        title
+        image
+      }
+    }
+  `;
+
+  beforeEach(async () => {
+    await deleteKeysByPattern('rateLimit:*');
+  });
+
+  const variables: Record<string, string> = {
+    sourceId: 's1',
+    url: 'https://daily.dev',
+  };
+
+  it('should not authorize when not logged in', () =>
+    testMutationErrorCode(
+      client,
+      {
+        mutation: MUTATION,
+        variables,
+      },
+      'UNAUTHENTICATED',
+    ));
+
+  it('should return link preview', async () => {
+    loggedUser = '1';
+
+    const sampleResponse = {
+      title: 'We updated our RSA SSH host key',
+      image:
+        'https://github.blog/wp-content/uploads/2021/12/github-security_orange-banner.png',
+    };
+
+    nock(postScraperOrigin)
+      .post('/preview', { url: variables.url })
+      .reply(200, sampleResponse);
+
+    const res = await client.mutate(MUTATION, { variables });
+
+    expect(res.errors).toBeFalsy();
+    expect(res.data.checkLinkPreview.title).toEqual(sampleResponse.title);
+    expect(res.data.checkLinkPreview.image).toEqual(sampleResponse.image);
+  });
+
+  it('should rate limit getting link preview by 5', async () => {
+    loggedUser = '1';
+
+    const sampleResponse = {
+      title: 'We updated our RSA SSH host key',
+      image:
+        'https://github.blog/wp-content/uploads/2021/12/github-security_orange-banner.png',
+    };
+
+    const mockRequest = () =>
+      nock(postScraperOrigin)
+        .post('/preview', { url: variables.url })
+        .reply(200, sampleResponse);
+
+    mockRequest();
+    const res1 = await client.mutate(MUTATION, { variables });
+    expect(res1.errors).toBeFalsy();
+    mockRequest();
+    const res2 = await client.mutate(MUTATION, { variables });
+    expect(res2.errors).toBeFalsy();
+    mockRequest();
+    const res3 = await client.mutate(MUTATION, { variables });
+    expect(res3.errors).toBeFalsy();
+    mockRequest();
+    const res4 = await client.mutate(MUTATION, { variables });
+    expect(res4.errors).toBeFalsy();
+    mockRequest();
+    const res5 = await client.mutate(MUTATION, { variables });
+    expect(res5.errors).toBeFalsy();
```

Comment:
I suggest looping instead of hard coding

Reply:
Good point! Coming right up!